TODO1/build-a-state-management-system-with-vanilla-javascript.md

```diff
@@ -2,42 +2,42 @@
 > * 原文作者：[ANDY BELL](https://css-tricks.com/author/andybell/)
 > * 译文出自：[掘金翻译计划](https://github.com/xitu/gold-miner)
 > * 本文永久链接：[https://github.com/xitu/gold-miner/blob/master/TODO1/build-a-state-management-system-with-vanilla-javascript.md](https://github.com/xitu/gold-miner/blob/master/TODO1/build-a-state-management-system-with-vanilla-javascript.md)
-> * 译者：
+> * 译者：[Shery](https://github.com/shery)
 > * 校对者：
 
-# Build a state management system with vanilla JavaScript
+# 使用原生 JavaScript 构建状态管理系统
 
-Managing state is not a new thing in software, but it’s still relatively new for building software in JavaScript. Traditionally, we’d keep state within the DOM itself or even assign it to a global object in the window. Now though, we’re spoiled with choices for libraries and frameworks to help us with this. Libraries like Redux, MobX and Vuex make managing cross-component state almost trivial. This is great for an application’s resilience and it works really well with a state-first, reactive framework such as React or Vue.
+状态管理在软件方面并不新鲜，但在 JavaScript 构建的应用中仍然相对较新。习惯上，我们会直接将状态保持在 DOM 上，甚至将其分配给 window 中的全局对象。但是现在，我们已经有了许多选择，这些库和框架可以帮助我们管理状态。像 Redux，MobX 和 Vuex 这样的库可以轻松管理跨组件状态。它大大提升了应用程序的适应能力，并且它对于状态优先的响应式框架（如 React 或 Vue）非常有用。
 
-How do these libraries work though? What would it take to write one ourselves? Turns out, it’s pretty straightforward and there’s an opportunity to learn some really common patterns and also learn about some useful modern APIs that are available to us.
+这些库是如何运作的？我们自己写个状态管理会怎么样？事实证明，它非常简单，并且有机会学习一些非常常见的设计模式，同时了解一些既有用又能用的现代 API。
 
-Before we get started, it’s recommended that you have an intermediary knowledge of JavaScript. You should know about data types and ideally, you should have a grasp of some more modern ES6+ JavaScript features. If not, [we’ve got your back](https://css-tricks.com/learning-gutenberg-4-modern-javascript-syntax/). It’s also worth noting that I’m not saying that you should replace Redux or MobX with this. We’re working on a little project to skill-up together and, hey, it could definitely power a small application if you were keeping an eye on the size of your JavaScript payload.
+在我们开始之前，请确保你已掌握中级 JavaScript 的知识。你应该了解数据类型，理想情况下，你应该掌握一些更现代的 ES6+ 语法特性。如果没有，[这可以帮到你](https://css-tricks.com/learning-gutenberg-4-modern-javascript-syntax/)。值得注意的是，我并不是说你应该用这个代替 Redux 或 MobX。我们正在一起开发一个小项目来提升技能，嘿，如果你一直关注 JavaScript 大小，它肯定能以最小代价为小型应用程序赋能。
```

Comment:
@IridescentMia 感谢校对
it could definitely power a small application if you were keeping an eye on the size of your JavaScript payload. 此处我的理解是作者想表达只需要添加少量代码就可以为应用添加新功能，在 http 传输中 payload 有有效数据的意思，就是传输中所携带的有效数据，JavaScript payload 我的理解是通过网络获取的 JavaScript 文件，JavaScript 文件本身的大小也是一种开销，如果关注网页性能的话 JavaScript 文件大小也是一个重要指标。

Reply:
我是这么理解的 原文是说

关注下js的状态数据的规模大小，如果是小型应用，用文中的状态管理是肯定没问题的。

作者的源代码函数中出现了参数payload，没记错的话，应该是传的状态数据。

要不问问 @coconilu 怎么看？